src/library/dao/macrodao.cpp

```diff
@@ -0,0 +1,111 @@
+#include "library/dao/macrodao.h"
+
+#include <QDebug>
+#include <QSqlQuery>
+
+#include "library/queryutil.h"
+
+bool MacroDAO::saveMacro(TrackId trackId, Macro* macro, int slot) const {
+    QSqlQuery query(m_database);
+    if (macro->getId() == -1) {
+        query.prepare(QStringLiteral(
+                "INSERT INTO macros "
+                "(track_id, slot, label, state, content) "
+                "VALUES "
+                "(:track_id, :slot, :label, :state, :content)"));
+        if (slot == 0) {
```

Comment:
```suggestion
        if (slot == kAutoAssignNextFreeMacroSlot) {
```

But this doesn't belong into MacroDAO IMHO! Also, this is dead code until we have channel-wide recording controls, right? Please remove if this is correct.